metron-platform/elasticsearch-shaded/pom.xml

```diff
@@ -31,7 +43,7 @@
         </dependency>
         <dependency>
             <groupId>org.elasticsearch.client</groupId>
-            <artifactId>transport</artifactId>
+            <artifactId>x-pack-transport</artifactId>
```

Comment:
Unlike the plain transport client, this jar is NOT Apache licensed.  I pulled it down and it has this
```
  <artifactId>x-pack-transport</artifactId>
  <version>5.6.7</version>
  ...
  <licenses>
    <license>
      <name>Elastic Commercial Software End User License Agreement</name>
      <url>https://www.elastic.co/eula/</url>
      <distribution>repo</distribution>
    </license>
  </licenses>
```

Reply:
For posterity, I pulled the jar with
```
{11:23}~ ➭ mvn org.apache.maven.plugins:maven-dependency-plugin:3.0.2:get \
    -DremoteRepositories=https://artifacts.elastic.co/maven \
    -Dartifact=org.elasticsearch.client:x-pack-api:5.6.7
```
and just opened up
`~/.m2/repository/org/elasticsearch/client/x-pack-transport/5.6.7/x-pack-transport-5.6.7.pom`

If someone knows more about how the ES licensing for this stuff works, let me know, because it' s definitely possible there's something I don't know about.

Reply:
I believe this is dual licensed ES commercial and ASLv2.  IT comes from [here](https://github.com/elastic/elasticsearch#license).

Reply:
The main Elastic stuff is (e.g. the plain transport client if you pull it down has Apache license), but the x-pack stuff doesn't seem to. The elastic/elasticsearch repo (from a quick search) does not contain the x-pack code from what I can tell.

Elastic's blog post from the other day also seems to imply it's not in there (and not ASLv2), but I could be reading too far into it. Check out https://www.elastic.co/products/x-pack/open

>We are creating a new X-Pack folder in each of these repositories that will be licensed under the Elastic EULA, which allows for some derivative works and contribution.

Reply:
That project does not appear to contain `PreBuiltXPackTransportClient`. ES points users to this - https://www.elastic.co/guide/en/elasticsearch/reference/6.2/setup-xpack-client.html - which instructs users to add the ES Maven repo. The referenced jar's license file does not appear to dual license as Apache V2. https://artifacts.elastic.co/maven/org/elasticsearch/client/x-pack-transport/6.2.2/x-pack-transport-6.2.2.jar

Reply:
Yep, I concede the point, that is not appropriately licensed via apache and we cannot bundle that dependency.